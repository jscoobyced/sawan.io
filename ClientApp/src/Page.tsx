import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { CandleChartPageHoc } from './components/charts/CandleChartPageHoc';
import { About } from './components/main/About';
import { BlogHoc } from './components/main/BlogHoc';
import { HomeHoc } from './components/main/HomeHoc';
import { Layout } from './components/main/Layout';
import { Resume } from './components/main/Resume';
import { ContentServiceFactory } from './services/ContentServiceFactory';
import { IContentService } from './services/IContentService';
import { Language, MainContent } from './services/Models';
import { HtmlUtils } from './utils/HtmlUtils';

export interface PageState {
    allContent: MainContent;
}

export class Page extends React.Component<{}, PageState> {

    private readonly contentService: IContentService;

    public constructor(props: {}) {
        super(props);
        this.contentService = ContentServiceFactory.GetContentService();
        this.state = {
            allContent: this.contentService.getDefaultMainContent()
        };
    }

    public componentDidMount() {
        this.contentService.getMainContent(Language.English)
            .then(allContent => {
                this.setState({ allContent });
            });
    }

    public render() {
        if (HtmlUtils.queryString().indexOf('full') > 0) {
            // hard-coded resume. Will need something more dynamic
            return <Resume />;
        }
        return <BrowserRouter children={this.routes()} basename={HtmlUtils.baseUrl()} />;
    }

    private routes() {
        return <Layout allContent={this.state.allContent} >
            <Route exact path='/' component={HomeHoc} />
            <Route path='/candle' component={CandleChartPageHoc} />
            <Route path='/health' component={About} />
            <Route path='/blog/:id' component={BlogHoc} />
            <Route path='/about' component={About} />
            <Route path='/resume' component={Resume} />
        </Layout>;
    }
}
