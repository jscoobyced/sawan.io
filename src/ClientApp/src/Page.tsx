import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { CandleChartPageHoc } from './components/charts/CandleChartPageHoc';
import { Home } from './components/main/Home';
import { Layout } from './components/main/Layout';
import { ContentServiceFactory } from './services/ContentServiceFactory';
import { IContentService } from './services/IContentService';
import { BlogPage, Language, MainContent } from './services/Models';
import { HtmlUtils } from './utils/HtmlUtils';

export interface PageState {
    allContent: MainContent;
    blogPage: BlogPage;
}

export class Page extends React.Component<{}, PageState> {

    private readonly contentService: IContentService;

    public constructor(props: any) {
        super(props);
        const mode = process.env.mode as string;
        this.contentService = ContentServiceFactory.GetContentService(mode);
        this.state = {
            allContent: this.contentService.getDefaultMainContent(),
            blogPage: this.contentService.getDefaultBlogPage()
        };
    }

    public componentDidMount() {
        this.contentService.getMainContent(Language.English)
            .then(allContent => {
                this.contentService.getBlogPage(0, 4)
                    .then(blogPage => {
                        this.setState({ allContent, blogPage });
                    });
            });
    }

    public render() {
        const routes = <Layout allContent={this.state.allContent} >
            <Route exact path='/' component={() => <Home blogPage={this.state.blogPage} />} />
            <Route path='/candle' component={CandleChartPageHoc} />
            <Route path='/health' component={CandleChartPageHoc} />
        </Layout>;
        return <BrowserRouter children={routes} basename={HtmlUtils.baseUrl()} />;
    }
}
