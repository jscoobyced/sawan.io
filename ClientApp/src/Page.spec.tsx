import { shallow } from 'enzyme';
import React from 'react';
import { Page } from './Page';

test('Page content is preserved', () => {
  const page = shallow(<Page />);
  expect(page.find('BrowserRouter')).toHaveLength(1);
  expect(page.find('BrowserRouter').children()).toHaveLength(1);
  expect(page.find('Route')).toHaveLength(6);
});
