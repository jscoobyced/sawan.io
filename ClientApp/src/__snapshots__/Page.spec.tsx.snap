// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page content is preserved 1`] = `
<BrowserRouter
  basename=""
>
  <Layout
    allContent={
      Object {
        "footerContent": Object {
          "copyright": "",
          "credits": "",
          "year": "",
        },
        "menuContent": Object {
          "links": Array [],
        },
        "navigationMenuContent": Object {
          "about": "",
          "applications": "",
          "cryptoCurrency": "",
          "healthMonitor": "",
          "home": "",
          "information": "",
          "resume": "",
          "websiteName": "",
        },
      }
    }
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/candle"
    />
    <Route
      component={[Function]}
      path="/health"
    />
    <Route
      component={[Function]}
      path="/blog/:id"
    />
    <Route
      component={[Function]}
      path="/about"
    />
    <Route
      component={[Function]}
      path="/resume"
    />
  </Layout>
</BrowserRouter>
`;
